### PR TITLE
Update README of azure-identity-providers-jdbc-[mysql/postgresql] libraries

### PR DESCRIPTION
**Pull request by backwind1233**
Make some improvements:
- Update README of azure-identity-providers-jdbc-mysql
- Update README of azure-identity-providers-jdbc-postgresql

### Docs
[azure-storage-blob](https://github.com/Azure/azure-sdk-for-java/tree/8ec2bcfa43c90aa5e716b8cf1ac212e9b8000cdc/sdk/storage/azure-storage-blob)
[azure-identity](https://github.com/Azure/azure-sdk-for-java/tree/8ec2bcfa43c90aa5e716b8cf1ac212e9b8000cdc/sdk/identity/azure-identity).

### CODE REVIEW

**Comment by backwind1233** [comment]
@stliu Could you help review this PR?

**Comment by saragluna** [comment]
@backwind1233, I have to change the target branch so I force-pushed to your PR. Please do not pull directly.

**Comment by stliu** [comment]
do we need a README in `sdk/jdbc` folder as https://github.com/backwind1233/azure-sdk-for-java/tree/update-jdbc-readme/sdk/keyvault ?

**Review by stliu — APPROVED** [review]
There is still room to improve this doc I think
for example, in the provider-core module, we should introduce the background and concept "token-as-password", and the API we provide in this module and how to use this api.

something I'd expect:

this module is an extension of azure-identity, it's main functionality is providing an template for the token-as-password secenrio

what is token-as-password?
Most azure serivces support authention with Azure AD natively, but there are some services, mostly OSS ones do not aware of Azure AD existance nor have an extensible auth mechanism but require a password. 
But we consider having password is against zero-trust principal, (see password-less for more details THE DOC / LINK).
to make the adoption more easily, we provide this token-as-password template

balabala

File: sdk/jdbc/azure-identity-providers-jdbc-mysql/README.md [hunk 1]
```diff
@@ -2,17 +2,24 @@
 
 This package contains the jdbc authentication plugin to authenticate with Azure AD for Azure hosted MySQL services.
 
+
+[Source code][source] | [API reference documentation][docs] | [Product documentation][product_docs] | [Quick_start][quick_start_mysql]
```
Comment (saragluna) [inline sdk/jdbc/azure-identity-providers-jdbc-mysql/README.md:6]:
```suggestion
[Source code][source] | [API reference documentation][docs] | [Product documentation][product_docs] | [Quickstart][quick_start_mysql]
```

File: sdk/jdbc/azure-identity-providers-jdbc-postgresql/README.md [hunk 2]
```diff
@@ -1,17 +1,22 @@
 # Azure identity JDBC PostgreSQL plugin library for Java
 
 This package contains the jdbc authentication plugin to authenticate with Azure AD for Azure hosted PostgreSQL services.
+[Source code][source] | [API reference documentation][docs] | [Product documentation][product_docs] | [Quick_start][quick_start_postgresql]
```
Comment (saragluna) [inline sdk/jdbc/azure-identity-providers-jdbc-postgresql/README.md:4]:
```suggestion
[Source code][source] | [API reference documentation][docs] | [Product documentation][product_docs] | [Quickstart][quick_start_postgresql]
```

**Comment by backwind1233** [comment]
Closing this PR, redirect to https://github.com/Azure/azure-sdk-for-java/pull/31380.

**Pull request closed without merging**